multiqc/modules/vep/vep.py

```diff
@@ -0,0 +1,221 @@
+            if len(line) < 1:
+                continue
+            if line[0] == "[":
+                title = line.replace("[", "").replace("]", "")
+                txt_data[title] = {}
+                continue
+            key, value = line.split("\t")
+            if key == "Novel / existing variants":
+                values = value.split("/")
+                novel = values[0].split("(")[0].replace(" ", "")
+                existing = values[1].split("(")[0].replace(" ", "")
+                txt_data[title]["Novel variants"] = int(novel)
+                txt_data[title]["Existing variants"] = int(existing)
+            elif title != "VEP run statistics":
+                txt_data[title][key] = int(value)
+            else:
+                txt_data[title][key] = value
+
+        for title in txt_data.keys():
+            if title not in self.vep_data[f["s_name"]]:
+                self.vep_data[f["s_name"]][title] = {}
+            self.vep_data[f["s_name"]][title] = txt_data[title]
+
+    def add_stats_table(self):
+        """Add a section with VEP General Statistics"""
+        table_config = {
+            "id": "vep-general-stats",  # ID used for the table
+            "table_title": "VEP General Statistics",  # Title of the table. Used in the column config modal
+            "save_file": False,  # Whether to save the table data to a file
+            "sortRows": True,  # Whether to sort rows alphabetically
+            "only_defined_headers": False,  # Only show columns that are defined in the headers config
+            "col1_header": "Sample Name",  # The header used for the first column
+            "no_beeswarm": True,  # Force a table to always be plotted (beeswarm by default if many rows)
+        }
+        title = "General statistics"
+        plot_cats = OrderedDict()
+        plot_data = OrderedDict()
+        color_list = ["Oranges", "Blues", "Reds", "Greens"]
+        for sample_name in self.vep_data:
+            if self.vep_data[sample_name][title] and len(self.vep_data[sample_name][title]) > 0:
+                order = 0
+                plot_data[sample_name] = {}
+                for key in self.vep_data[sample_name][title]:
+                    order += 1
+                    if key not in plot_cats:
+                        plot_cats[key] = {"name": key, "format": "{:,.0f}", "scale": color_list[order % 4]}
+                    plot_data[sample_name][key] = self.vep_data[sample_name][title][key]
+        if len(plot_data) > 0:
+            self.add_section(
+                name="General Statistics",
+                anchor="vep-general-statistics",
+                description="VEP General statistics",
+                helptext="Table showing general statistics of VEP annotaion run",
+                plot=table.plot(plot_data, plot_cats, table_config),
+            )
+
+    def generate_bar_graphs(self):
+        """Add bar graphs from the given VEP stats"""
+        titles = [
```

Comment:
Whilst it's less code and cleaner (and DRY) to write the bar graphs in a loop like this, I prefer not to do this for MultiQC modules. The best MultiQC modules have well polished output, which basically comes down to fine-tuning how each plot looks and reads. This kind of customisation is very awkward to do if in a loop.

The two immediate examples here are that the descriptions are pretty unhelpful, and I want to customise the colours on one bar plot. This super tricky straight away.